drivers/digitalocean/digitalocean.go

```diff
@@ -297,18 +297,22 @@ func (d *Driver) Create() error {
 				d.PrivateIPAddress = network.IPAddress
 			}
 		}
+		if d.IPv6 {
+			d.IPv6Address, _ = newDroplet.PublicIPv6()
+		}
 
-		if d.IPAddress != "" && (!d.PrivateNetworking || d.PrivateIPAddress != "") {
+		if d.IPAddress != "" && (!d.PrivateNetworking || d.PrivateIPAddress != "") && (!d.IPv6 || d.IPv6Address != "") {
```

Comment:
Just a general question, but can the `PrivateIPAddress` be an ipv6 address? does it matter?